Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -311,6 +313,20 @@ private func createResolvedPackages(
             }
         }
     }
+    
+    // If a target with similar name was encountered before, we emit a diagnostic.
+    if foundDuplicateTarget {
+        for targetName in allTargetNames.sorted() {
+            // Find the packages this target is present in.
+            let packageNames = packageBuilders
+                .filter({ $0.targets.contains(where: { $0.target.name == targetName }) })
+                .map({ $0.package.name })
+                .sorted()
+            if packageNames.count > 1 {
```

Comment:
nit: We can use `packageNames.isEmpty` instead.

Reply:
I don't think so, since it needs to be `> 1`  not just `!isEmpty`, right?